docs/gke-installation.md

```diff
@@ -0,0 +1,126 @@
+# Deploying Antrea on a GKE cluster
+
+We support running Antrea inside of GKE clusters on Ubuntu host. Antrea would operate
+in noEncap mode, in which no encapsulation is required for any kind of traffic (Intra Node,
+Inter Node etc) and enforcing network policies using OVS.
+
+## GKE Prerequisites
+
+1. Install the Google Cloud SDK (gcloud). Refer to this (https://cloud.google.com/sdk/install)
+
+    ```bash
+    curl https://sdk.cloud.google.com | bash
+    ```
+
+2. Make sure you are authenticated to use the Google Cloud API
+
+    ```bash
+    export ADMIN_USER=user@email.com
+    gcloud auth login
+    ```
+
+3. Create a project or use an existing one
+
+    ```bash
+    export GKE_PROJECT=gke-clusters
+    gcloud projects create $GKE_PROJECT
+    ```
+
+## Creating the cluster
+
+You can use any method to create a GKE cluster (gcloud SDK, gcloud Console etc). The example
+given here is using the Google Cloud SDK.
+
+**Note:** Antrea is supported on Ubuntu hosts only for GKE cluster. Also, its must to select service
+CIDR at the time of cluster deployment.
+
+1. Create a GKE cluster
+
+    ```bash
+    export GKE_ZONE="us-west1"
+    export GKE_HOST="UBUNTU"
+    export GKE_SERVICE_CIDR="10.94.0.0/16"
+    gcloud container --project $GKE_PROJECT clusters create cluster1 --image-type $GKE_HOST \
+       --zone $GKE_ZONE --enable-ip-alias --services-ipv4-cidr $GKE_SERVICE_CIDR
+    ```
+
+2. Access your cluster
+
+    ```bash
+    kubectl get nodes
+    NAME                                      STATUS   ROLES    AGE     VERSION
+    gke-cluster1-default-pool-93d7da1c-61z4   Ready    <none>   3m11s   v1.14.10-gke.17
+    gke-cluster1-default-pool-93d7da1c-rkbm   Ready    <none>   3m9s    v1.14.10-gke.17
+    ```
+
+3. Create a cluster-admin ClusterRoleBinding
+
+    ```bash
+    kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user user@email.com
+    ```
+
+    **Note:** To create clusterRoleBinding, the user must have `container.clusterRoleBindings.create` permission.
+Use this command to enable it, if the previous command fails due to permission error. Only cluster Admin can
+assign this permission.
+
+    ```bash
+    gcloud projects add-iam-policy-binding $GKE_PROJECT --member user:user@email.com --role roles/container.admin
+    ```
+
+## Deploying Antrea
+
+1. Prepare the Cluster Nodes
+
+    Deploy antrea-node-init DaemonSet to enable kubelet to operate in CNI mode.
+
+    ```bash
+    kubectl apply -f https://github.com/vmware-tanzu/antrea/releases/download/v0.5.0/patches/gke/antrea-node-init.yml
```

Comment:
@antoninbas @tnqn @abhiraut Will this be a valid link ??? Please check.

Reply:
@antoninbas , Is there any possible we have a latest tag of release? Thus people could always get newest release from the same URL, like ` https://github.com/vmware-tanzu/antrea/releases/download/latest/patches/gke/antrea-node-init.yml`.

Reply:
Please check ipsec-tunnel.md and eks-installation.md to describe yaml files.

Reply:
Unfortunately, there is no link "alias" for the latest published release.

@reachjainrahul this link will definitely *not* work. There is no directory structure for assets, so *if* we wanted to have this as an asset, the link would have to be something like `https://github.com/vmware-tanzu/antrea/releases/download/v0.5.0/antrea-gke-node-init.yml`. However, there is no reason to publish this as an asset here: the reason why we publish Antrea yaml manifests as assets is because the release version is different from the dev version (e.g. we use a different Antrea docker image tag). In the case of antrea-node-init.yml, there is a single version of the yaml (it's pretty much never going to change), so you should use a "regular github tree link": based on the current location, it will be `https://github.com/vmware-tanzu/antrea/tree/v0.5.0/build/yamls/patches/gke/antrea-node-init.yml`. However, I don't think that placing it under "patches" is appropriate here.

Reply:
Yes we will publish the yaml from 0.5.0. But here I suggest you to follow description of other yaml files on how to get the yaml - link for the latest one, and that for a released version. You might consider some reorganization of your descriptions to talk about released versions only once for antrea-gke-node-init.yml and antrea-gke.yml.

Reply:
I am keeping antrea-gke-node-init documentation as is and updated antrea-gke.yml documentation to reflect latest and released version. @antoninbas I hope this link would work.  https://github.com/vmware-tanzu/antrea/releases/download/v0.5.0/antrea-gke-node-init.yml